### PR TITLE
205 BE: add room resource

### PR DESCRIPTION
**Pull request by esteban-gs**
The ticket for this PR requires creating a `Rom` entity, but to find the simplest path forward to Beta, I am adapting the `Transaction` entity to serve as the table that tracks the chat rooms that should be provisioned, so this PR is proposing to use `Transactions` as a chat room

This ticket was conceived following these assumptions:
- When a user (org agent) claims an `Asset` (posted by a user), a new transaction is created which serves as the `Room`
- Each `Asset`, when posted does not need a corresponding transaction record to exist
- The Room is a private channel with all the parties involved. Primarily, the `Asset` poster (who will appear as the `donater_user` under transactions), the claimer, who is the user who activated the claim action
- The user claiming the `Asset` can only be a user who is an agent for an org
- There will be a number of transactions open for each `Asset`, which will be the `"chat rooms"` where people can decide on their own. 

### Other changes in the general spirit of this ticket
I renamed a few fields in the Transactions entity to make it more clear what each does. I added a `claimed_transactions` relationship from Orgs to Transactions to differentiate from the donated transactions.

Changed a bad relationship type for `Transactions.messages`

### repro steps:
- run app
- create:
  - `Asset`
  - `User`
  - `Organization`
- create a new transaction:
  - with `JSON`  
```json
{
	"donater_user": {"id":1},
	"asset": {"id":1},
	"claimer": {"id":1}
}
```
  - at http://localhost:3001/api/transactions

Expect:
`201 Create` status response
with `json` body like:
```json
{
  "donater_user": {
    "id": 1
  },
  "asset": {
    "id": 1
  },
  "claimer": {
    "id": 1
  },
  "id": 2,
  "status": "NEW_CLAIM",
  "created_date": "2022-10-12T09:15:41.394Z"
}
```

### CODE REVIEW

File: server/src/main.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ async function bootstrap() {
     // TODO get env related base url
     origin: 'http://localhost:3000',
   });
-  app.useGlobalPipes(new ValidationPipe());
+  app.useGlobalPipes(new ValidationPipe({ transform: true }));
```
Comment (esteban-gs) [inline server/src/main.ts:20]: This allows for type-safety in controller method parameters

File: server/src/organizations/entities/organization.entity.ts [hunk 2]
```diff
@@ -41,9 +41,9 @@ export class Organization {
...
-    () => Transaction,
-    (transaction) => transaction.donater_organization || transaction.recipient,
-  )
-  transactions: Transaction[];
+  @OneToMany(() => Transaction, (transaction) => transaction.donater_organization)
+  donated_transactions: Transaction[];
+
+  @OneToMany(() => Transaction, (transaction) => transaction.claimer)
+  claimed_transactions: Transaction[];
 }
```
Comment (esteban-gs) [inline server/src/organizations/entities/organization.entity.ts:49]: Split the two to make them more explicit. One tracks donated stuff, the other claimed stuff

File: server/src/transactions/dto/create-transaction.dto.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export class CreateTransactionDto {
   @IsOptional()
   donater_organization?: Organization;
 
-  @IsNotEmpty({ message: 'asset_id is required' })
+  @IsNotEmpty({ message: 'asset is required' })
```
Comment (esteban-gs) [inline server/src/transactions/dto/create-transaction.dto.ts:17]: just keeping consistent with existing code

**Review by jd2rogers2 — APPROVED** [review]
- [ ] one copy pasta change (see inline comment)

then LGTM!! thanks!
good catch on the transactions.messages field ⚾

File: server/src/transactions/entities/transaction.entity.ts [hunk 4]
```diff
@@ -32,19 +35,19 @@ export class Transaction {
...
+  @ManyToOne(() => Organization, (organization) => organization.donated_transactions)
   @JoinColumn()
   donater_organization?: Organization;
 
   @ManyToOne(() => Asset, (asset) => asset.transactions, { eager: true })
   @JoinColumn()
   asset: Asset;
 
-  @ManyToOne(() => Organization, (recipient) => recipient.transactions)
+  @ManyToOne(() => Organization, (org) => org.donated_transactions)
```
[the remark this answers is not in the document]
Comment (esteban-gs) [reply server/src/transactions/entities/transaction.entity.ts:46]: Good catch!!

**Pull request merged**